src/Microsoft.PowerApps.TestEngine/PowerFx/Visitors/DetermineNearestFormulaSeparationForSpansVisitor.cs

```diff
@@ -0,0 +1,248 @@
+﻿// Copyright (c) Microsoft Corporation.
```

Comment:
Does this logic exist in the latest versions of PowerFx?
Are we doing this here because the current version of PowerFx used in TestEngine does not have this logic?

Reply:
No this doesn't exist there. it should be in there tho. powerfx exposes base visitor classes to allow its consumers like TE to write any custom or domain specific logic that requires traversing through the parse tree. it doesn't have a use case where it needs to find formulas separated by semicolon at the top most level and evaluating each formula. this is TE specific.